File: kotlin/src/main/kotlin/org/sammancoaching/phonebook/BadPhonebookEntryEvent.kt
package org.sammancoaching.phonebook

@JvmRecord
data class BadPhonebookEntryEvent(
    val phonebookSize: Int, val name: String, val number: String,
    val clashingEntry: Pair<String?, String?>
) {
    fun eventToAlert(): AlertData {
        return AlertData(
            phonebookSize.toString(),
            "$name cannot be added with number $number",
            clashingEntry.first + " with number " + clashingEntry.second
        )
    }
}

File: csharp/EnterprisePhonebook-TestDesign-Kata/EnterprisePhonebook/BadPhonebookEntryEvent.cs
namespace SammanCoaching.Phonebook;

public sealed record BadPhonebookEntryEvent(
    int PhonebookSize,
    string Name,
    string Number,
    KeyValuePair<string, string> ClashingEntry)
{
    public AlertData ToAlertData() => new(
        PhonebookSize.ToString(),
        $"{Name} cannot be added with number {Number}",
        $"{ClashingEntry.Key} with number {ClashingEntry.Value}");
}
